tycho-common/src/dto.rs

```diff
@@ -1322,6 +1322,121 @@ impl ProtocolSystemsRequestResponse {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug, Default, PartialEq, ToSchema, Eq, Hash, Clone)]
+pub struct TracedEntryPointRequestBody {
+    #[serde(default)]
+    pub chain: Chain,
+    /// Filters by protocol, required to correctly apply unconfirmed state from
+    /// ReorgBuffers
+    pub protocol_system: String,
+    /// Filter by component ids
+    pub component_ids: Option<Vec<String>>,
+    /// Max page size supported is 100
+    #[serde(default)]
+    pub pagination: PaginationParams,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone, ToSchema, Eq, Hash)]
+pub struct EntryPoint {
+    #[schema(example = "0xEdf63cce4bA70cbE74064b7687882E71ebB0e988:getRate()")]
+    /// Entry point id.
+    pub external_id: String,
+    #[schema(value_type=String, example="0x8f4E8439b970363648421C692dd897Fb9c0Bd1D9")]
+    #[serde(with = "hex_bytes")]
+    /// The address of the contract to trace.
+    pub target: Bytes,
+    #[schema(example = "getRate()")]
+    /// The signature of the function to trace.
+    pub signature: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone, ToSchema, Eq, Hash)]
+pub struct RPCTracerParams {
+    #[schema(value_type=Option<String>)]
+    #[serde(with = "hex_bytes_option", default)]
+    pub caller: Option<Bytes>,
+    #[schema(value_type=String, example="0x679aefce")]
+    #[serde(with = "hex_bytes")]
+    pub calldata: Bytes,
+}
+
+impl From<models::blockchain::RPCTracerParams> for RPCTracerParams {
+    fn from(value: models::blockchain::RPCTracerParams) -> Self {
+        RPCTracerParams { caller: value.caller, calldata: value.calldata }
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq, Clone)]
+#[serde(tag = "method", rename_all = "lowercase")]
+pub enum TracingParams {
+    /// Uses RPC calls to retrieve the called addresses and retriggers
+    RPCTracer(RPCTracerParams),
+}
+
+impl From<models::blockchain::TracingParams> for TracingParams {
+    fn from(value: models::blockchain::TracingParams) -> Self {
+        match value {
+            models::blockchain::TracingParams::RPCTracer(params) => {
+                TracingParams::RPCTracer(params.into())
+            }
+        }
+    }
+}
+
+impl From<models::blockchain::EntryPoint> for EntryPoint {
+    fn from(value: models::blockchain::EntryPoint) -> Self {
+        EntryPoint {
```

Comment:
Rusty picky but I think it's nicer to use `Self` instead of EntryPoint for the return type. Avoids repeating the type name and if the name changes is one less update :)
(I've seen both patterns on the code, we don't enforce it, it's just a suggestion)